flink-libraries/flink-table/src/test/scala/org/apache/flink/table/ExpressionReductionTest.scala

```diff
@@ -422,4 +423,50 @@ class ExpressionReductionTest extends TableTestBase {
     util.verifyTable(result, expected)
   }
 
+  @Test(expected = classOf[NullPointerException])
+  def testReduceDeterministicUDF(): Unit = {
```

Comment:
The exception isn't caused by Flink, to check an unexpected exception is not a good way. I suggest to mark this test as `@Ignore` and add a TODO comment on the top of the test. After CALCITE-1860 is fixed, we can reopen this test or check&remove this test.